src/roller/table.ts

```diff
@@ -38,40 +43,148 @@ export class TableRoller extends GenericFileRoller<string> {
+
+                // Update tooltip
+                if (subRoller instanceof TableRoller) {
+                    subTooltips.push(subRoller.combinedTooltip);
+                }
+                else {
+                    const [top, bottom] = subRoller.tooltip.split("\n");
+                    subTooltips.push(top + " --> " + bottom);
+                }
+            }
+        }
+
+        res.combinedTooltip = subTooltips.join(";");
+
+        return res;
+    }
+
     async getResult() {
-        if (this.isLookup) {
-            const result = await this.lookupRoller.roll();
-            const option = this.lookupRanges.find(
-                ([range]) =>
-                    (range[1] === undefined && result === range[0]) ||
-                    (result >= range[0] && range[1] >= result)
-            );
-            if (option) {
-                return option[1];
+        let res = [];
+
+        let subTooltips: string[] = [];
+
+        for (let i = 0; i < this.rolls; i++) {
+            let subTooltip:string;
+            let subResult: SubRollerResult;
+            let selectedOption:string;
+
+            if (this.isLookup) {
+                const result = await this.lookupRoller.roll();
+                const option = this.lookupRanges.find(
+                    ([range]) =>
+                        (range[1] === undefined && result === range[0]) ||
+                        (result >= range[0] && range[1] >= result)
+                );
+                if (option) {
+                    subTooltip = this.lookupRoller.original.trim() + " --> " + `${this.lookupRoller.resultText}${this.header ? " | " + this.header : ""}`.trim();
+                    selectedOption = option[1];
+                }
+            }
+            else {
+                const options = [...this.options];
+                const randomRowNumber = this.getRandomBetween(0, options.length - 1);
+                // TODO: to be confirmed (was this to forbid rolling the same result twice ?)
+                // options.splice(options.indexOf(option), 1);
+                subTooltip = options.length + " rows" + " --> " + "[row " + (randomRowNumber+1) + "]";
+                selectedOption = options[randomRowNumber];
+            }
+
+            subResult = await this.getSubResult(selectedOption);
+            res.push(subResult.result);
+
+            if (subResult.combinedTooltip) {
+                subTooltip += " > (" + subResult.combinedTooltip + ")";
             }
+            subTooltips.push(subTooltip);
         }
-        const options = [...this.options];
-
-        return [...Array(this.rolls)]
-            .map(() => {
-                let option =
-                    options[this.getRandomBetween(0, options.length - 1)];
-                options.splice(options.indexOf(option), 1);
-                return option;
-            })
-            .join("||");
+
+        // TODO: find a simpler way
```

Comment:
Hum, not sure that nested arrays are simpler!

I suppose that you would have an array for each "level", but then... don't forget that each level would need potentially a different "join character".

I'm not sure I see what you have in mind.

Reply:
As for the looping per character, this is where I think it could be "optimized" using some regex to "consume" all characters that are of no interest for the formatting in one shot).

Reply:
OK, so I tried using regex... not sure it looks way better though 😁

```
prettify(input: string): string {
        const specialChars = /(.*?)(\(|\)|;|\|\|)(.*)/
        const tab = "\t";
        let tabCount = 0;
        let output:string = "";

        let remaining:string = input;
        let matches: RegExpMatchArray;
        while (matches = remaining.match(specialChars)) {
            let [, beforeSpecial, special, afterSpecial] = matches;
            output += beforeSpecial;
            if (special == ")") {
                tabCount--;
                output += "\n";
                output += tab.repeat(tabCount);
                output += ")";
            }
            else {
                if (special == "(") {
                    tabCount++;
                    output += "(";
                }
                else if (special == ";") {
                    output += ",";
                }
                else if (special == "||") {
                    output += "|";
                }
                output += "\n"
                output += tab.repeat(tabCount);
            }
            remaining = afterSpecial;
        }
        output += remaining;

        return output;
    }
```